File: java/week01CircularBarn.java
import java.util.Scanner;

public class week01CircularBarn {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        
        int[] destinations = new int[n];
        for (int i = 0; i < n; i++) {
            destinations[i] = scanner.nextInt();
        }

        int low = 100 * 1000;

        for (int i = 0; i < destinations.length; i++) {
            int total = 0;
            for (int j = 0; j < destinations.length; j++) {
                int destination = destinations[j];
                total += ((j - i + n) % n) * destination;
            }

            if (total < low) {
                low = total;
            }
        }

        System.out.println(low);

        scanner.close();
    }
}

File: python/week_01_circular_barn.py
n = int(input())
destinations = [int(input()) for x in range(n)]

low = 100 * 1000

for i in range(n):
    total = 0
    for j, destination in enumerate(destinations):
        total += ((j - i) % n) * destination
    low = min(low, total)

print(low)
